homeassistant/components/weatherflow/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""Get data from Smart Weather station via UDP."""
+from __future__ import annotations
+
+import logging
+
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.device import EVENT_LOAD_COMPLETE, WeatherFlowDevice
+from pyweatherflowudp.errors import ListenerError
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP, Platform
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.start import async_at_started
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [
+    Platform.SENSOR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WeatherFlow from a config entry."""
+
+    client = WeatherFlowListener()
+
+    @callback
+    def _async_device_discovered(device: WeatherFlowDevice) -> None:
+        _LOGGER.debug("Found a device: %s", device)
+
+        @callback
+        def _async_add_device_if_started(device: WeatherFlowDevice):
+            async_at_started(
+                hass,
+                callback(
+                    lambda _: async_dispatcher_send(
+                        hass, f"{DOMAIN}_{entry.entry_id}_add_{SENSOR_DOMAIN}", device
```

Comment:
```suggestion
                        hass, f"{DOMAIN}_{entry.entry_id}_add", device
```

The dispatcher here shouldn't need to know about the sensor domain